ush/python/pygfs/task/archive_vars.py

```diff
@@ -0,0 +1,362 @@
+                         Empty dict {} if no additional variables needed
+        """
+        # EnKF-specific: Only these 3 ENSSTAT paths with MEMDIR='ensstat'
+        if 'enkf' in self.task_config.RUN:
+            template_specs = [
+                ('COMIN_ATMOS_ANALYSIS_ENSSTAT', 'COM_ATMOS_ANALYSIS_TMPL', {'MEMDIR': 'ensstat'}),
+                ('COMIN_ATMOS_HISTORY_ENSSTAT', 'COM_ATMOS_HISTORY_TMPL', {'MEMDIR': 'ensstat'}),
+                ('COMIN_SNOW_ANALYSIS_ENSSTAT', 'COM_SNOW_ANALYSIS_TMPL', {'MEMDIR': 'ensstat'})
+            ]
+        else:
+            # All other systems (GFS, GEFS, GCAFS) get common + grid-specific paths
+            template_specs = [
+                ('COMIN_ATMOS_ANALYSIS', 'COM_ATMOS_ANALYSIS_TMPL', {}),
+                ('COMIN_ATMOS_GENESIS', 'COM_ATMOS_GENESIS_TMPL', {}),
+                ('COMIN_ATMOS_HISTORY', 'COM_ATMOS_HISTORY_TMPL', {}),
+                ('COMIN_ATMOS_TRACK', 'COM_ATMOS_TRACK_TMPL', {}),
+                ('COMIN_CHEM_ANALYSIS', 'COM_CHEM_ANALYSIS_TMPL', {}),
+                ('COMIN_SNOW_ANALYSIS', 'COM_SNOW_ANALYSIS_TMPL', {}),
+                ('COMIN_OBS', 'COM_OBS_TMPL', {}),
+                ('COMOUT_ATMOS_TRACK', 'COM_ATMOS_TRACK_TMPL', {}),
+            ]
+            # Grid-specific paths
+            for grid in ["0p25", "0p50", "1p00"]:
+                com_key = f"COMIN_ATMOS_GRIB_{grid}"
+                template_specs.append((com_key, 'COM_ATMOS_GRIB_GRID_TMPL', {'GRID': grid}))
+
+            # GEFS-specific: Ensemble statistics path
+            if 'gefs' in self.task_config.RUN:
+                template_specs.append(('COMIN_ATMOS_ENSSTAT_1p00', 'COM_ATMOS_GRIB_GRID_TMPL', {'GRID': '1p00'}))
+
+        return template_specs
+
+    def _construct_com_paths(self, base_dict: Dict[str, str], template_specs: list) -> Dict[str, str]:
+        """Construct COM paths from template specifications.
+
+        This method takes template specifications and constructs the actual paths
+        by substituting template variables using base_dict updated with extra_vars.
+
+        Parameters
+        ----------
+        base_dict : Dict[str, str]
+            Base template substitution dictionary from _get_template_dict()
+        template_specs : list of tuples
+            List from _get_com_template_specs() containing specifications
+
+        Returns
+        -------
+        Dict[str, str]
+            Dictionary mapping COM variable names to resolved paths
+        """
+        com_paths = {}
+
+        for com_key, template_key, extra_vars in template_specs:
+            # Use base_dict directly, updated with any extra variables
+            tmpl_dict = {**base_dict, **extra_vars}
+
+            template = self.task_config.get(template_key, '')
+            com_paths[com_key] = Template.substitute_string(
+                template, TemplateConstants.DOLLAR_CURLY_BRACE,
+                lambda key: tmpl_dict.get(key, '')) if template else ''
```

Comment:
I'm not thrilled about using `substitute_string`.  This is a deprecated function.  You did something different for `stage_ic`.  Would that approach not work here?